internal_packages/keybase/lib/passphrase-popover.cjsx

```diff
@@ -1,25 +1,67 @@
 {React, Actions} = require 'nylas-exports'
 Identity = require './identity'
+PGPKeyStore = require './pgp-key-store'
 _ = require 'underscore'
+fs = require 'fs'
+pgp = require 'kbpgp'
 
 module.exports =
 class PassphrasePopover extends React.Component
   constructor: ->
-    @state = {passphrase: ""}
+    @state = {
+      passphrase: ""
+      placeholder: "PGP private key password"
+      classNames: "key-passphrase-input form-control"
+    }
 
   @propTypes:
     identity: React.PropTypes.instanceOf(Identity)
+    addresses: React.PropTypes.array
 
   render: ->
-    passphrase = @state
     <div className="passphrase-popover">
-      <input type="password" value={@state.passphrase} placeholder="PGP private key password" className="key-passphrase-input form-control" onChange={@_onPassphraseChange} />
-      <button className="btn btn-toolbar" onClick={ @_onDone }>Done</button>
+      <input type="password" value={@state.passphrase} placeholder={@state.placeholder} className={@state.classNames} onChange={@_onPassphraseChange} onKeyUp={@_onCarriageReturn} />
+      <button className="btn btn-toolbar" onClick={@_validatePassphrase}>Done</button>
     </div>
 
   _onPassphraseChange: (event) =>
     @setState
       passphrase: event.target.value
+      placeholder: "PGP private key password"
+      classNames: "key-passphrase-input form-control"
+
+  _onCarriageReturn: (event) =>
```

Comment:
Stupid little nit, but it'd be nice if this was `_onKeyUp` since it technically isn't only fired on carriage returns.